docs/specification.md

```diff
@@ -544,9 +537,9 @@ unit
 
 ```haskell
 data Proposal proposalMetadata = Proposal
-    { frozen_token_requirement :: Natural
-    -- ^ How many tokens should be frozen in order to vote on this proposal.
-    , metadata :: proposalMetadata
+    { metadata :: proposalMetadata
+    , proposer :: Address
```

Comment:
Can `proposer` be different from `SENDER`?

Reply:
`proposer` is always `SENDER` of `Propose` entrypoint, unless we allow `Operators` to do it on the behalf. But currently we do not take operators into account.

I have to add this field to keep track of who the proposer so that we can unfreeze his tokens when `flush`.

Reply:
It's ok to store `proposer` address in storage, but it shouldn't be passed to `propose` entrypoint because it's redundant and error-prone. We have to make an extra check (that `proposer` equals to `SENDER`) which costs some gas and can be accidentally forgotten. So `Proposal` that we pass to `propose` shouldn't have `proposer :: Address` in it. In storage we can store something like `(Proposal, Address)`, for example.

Reply:
Also I think it makes sense to return `proposer` address from `proposal_metadata`, so maybe we need two types: one without `proposer` and one with it.

Reply:
Oh, I see you already have `ProposeParams`. That's what I had in mind as well, please update the specification accordingly.

Reply:
Sorry for the confusion, Ivan. I've updated `Config`, and added `Proposal`, `Storage` types according to the current MR of propose-related impl of baseDAO.

I've added a few errors that are needed, use `ProposeParam` for `propose` entrypoint, and add some new description addressing the new errors and bounded value.